### ISSUE TITLE
Skyrim VR PC virtual walking issue with vive trackers

### ISSUE THREAD

**Issue by BrentW89** [pottedmeat7/OpenVR-WalkInPlace#13]
Great work on this driver so far! That said, are there any settings that work well with Skyrim VR PC and foot trackers? The game is amazing and uses the controller pad for smooth locomotion, so I'd think it would work. I see the foot steps registering in the graph so the tracking is working. However, the only way I can get movement in the game is to use the keyboard input setting (and fyi, you MUST click in the game window before playing or the keyboard input doesn't get recognized!). Anyway, this technically works, but the movement speed is very slow and is fixed rather than variable like when using the vive directional pads. 

I've tried many settings like turning WIP on an off, controller A, B, both. Tracker on & HMD off. Various thresholds. Nothing works even though my footsteps are being recognized in your graph. 

If keyboard is the only option for this game, then there at least needs to be a way specify a faster walking speed. Even if variable speed is not possible, if faster foot steps triggered sprint mode, that would be great for immersion (and for my goal of getting more exercise in an enjoyable way). I look forward to hearing from you! You're so close to giving us the most immersive Skyrim VR experience ever!

**Comment by BrentW89** [comment]
In follow up to my previous post, I tried uninstalling OpenVR Walk and reinstalling and it says that the product is still installed and do I want to upgrade it. You seem to have a bug where all the registry settings and other things don't get uninstalled (at least in my case). The reinstall did work though. I also tried reinstalling Skyrim VR for kicks, but that didn't help.

When I tried to install the 'OpenVR-WalkInPlace-OVRIE-driver.exe', it said I needed the Input Emulator product. However, I'm having difficulty finding where the setup.exe is for that product (I see source code only on GitHub), so if you have a link that would be helpful. Also, some guidance on what this product would do for me and others would be awesome.

**Comment by pottedmeat7** [comment]
I have also noticed some issue with Skyrim, seems like only the click to sprint virtual input is working when running is detected. I'll look into it unless its a change to OpenVR I don't know why it doesn't work only on Skyrim. 
You can get Input Emulator here https://github.com/matzman666/OpenVR-InputEmulator/releases.

**Comment by BrentW89** [comment]
Thanks! Can you clarify how Input emulator works with your driver and what it's useful for as far as your users are concerned?

Also, after only getting the tracker keyboard emulation to work, I installed the steam beta via their properties menu. Keyboard emulation still works, but now graphics are dramatically better on my GTX1080. I think steam is super sampling existing games based on your computer's capabilities (and it's suppose to scale resolution for the Vive Pro or other headsets). I got the best results when I slid the Skyrim super-sampling bar all the way to the left. I hope your next update will remain compatible with this beta and their next release and it significantly enhanced immersion.

**Comment by pottedmeat7** [comment]
OpenVR-Input Emulator will do the virtual input application for the back-end driver it has the same driver system my normal WalkInPlace installer has. The drivers are the same, so if you want both there's no reason to install two separate drivers. So basically the 'OpenVR-WalkInPlace-OVRIE-driver.exe' installer doesn't install any drivers and uses the input emulator driver.

**Comment by BrentW89** [comment]
In follow up to my tip #1, Steam appears have to implemented the graphical scaling into the official release, which is good. Regarding your VR walking emulator in keyboard mode, I love physically walking around small areas (slow though), but when pressing the directional pad to jog (while physicality stepping up and down), it will stop periodically requiring me to continuely re-touch the directional pad to get it going again in Skyrim. I assume this is a side-effect of the keyboard input.

Any progress on the variable speed directional pad support? Skyrim VR with mods is absolutely incredible. Real virtual walking/jogging seems so close!

**Comment by pottedmeat7** [comment]
@BrentW89 looks like there was just a slight difference in the touchpad input that Skyrim needed in order to correctly register. It should be working now try out the new release here https://github.com/pottedmeat7/OpenVR-WalkInPlace/releases .

**Comment by BrentW89** [comment]
This update is so much better! Is there an option to activate variable speed movement backed on how quick the tracker walking pace is?  If that were an option, I'd definitely be donating to your cause.

It took me a while to figure out I could swing my arms to trigger jog and run modes with the tracker. This works pretty well, but only after I fiddled with the numbers for awhile. That said, it can still create an unnatural lurching forward affect since it's not tied to tracker foot step speed. I've also noticed it temporarily stopping for no reason occasionally. That said, an arm movement mode makes sense if someone wanted to play without exercising their legs as much and really makes sense for sprinting.

Can you tell us what your recommended settings are for Skyrim VR with trackers? (I'm still not entirely certain I have the ideal settings and I'm sure others would find this useful). 

Suggestion: If you included default profiles for some major games that were at least close, you'd have far less confusion and people thinking it didn't work well. The docs could also be improved with more examples and clarity.

Also, your graph screen was confusing as I kept trying to adjust XYZ sensitivity based on the first graph and it had no effect until I realized I needed set much larger values close to 1.00 to allow me to fight NPC's or shuffle around the play space without triggering open walk.

**Comment by pottedmeat7** [comment]
Varying the speed on the pace of the trackers is on my list. I think that could work but it also will need some threshold of strength of movement I believe because smaller steps (walking) can be also pretty quick for instance a jog can actually be pretty slow comparatively. However jogging should cause stronger movements so it may need to be a combination of both. 
Not sure if I want to combine this into a bigger update to have an auto config system, but the UI will definitely need more configuration to change the pace, and strength, of a step for trackers. 

Also some of the other features you mention I will take into consideration and will plan what will work best.

**Comment by BrentW89** [comment]
I did some more testing in Skyrim VR with WIP and various touchpad settings, and the bug with it randomly starting and stopping movement is far more frequent than I first thought. It seems to happen randomly within every 5 to 30 seconds of movement. I then have to move faster, swing more or less, touch the pad, or something to get it "unstuck" and to start moving again. It's highly annoying. Thinking it might be a bug in the tracker implementation, I disabled that and enabled WIP, HMD, and arms. This works well with some tweaks but suffers from the same annoying stop and starts, especially when walking. 

Are you seeing this? If not, what might be causing it?  If yours works with walk, job, sprint, then a screen shot of your working Skyrim implementation with and without trackers would be very helpful!

**Comment by pottedmeat7** [comment]
You are probably experience the issue with Running with click to sprint. I think again there is a different method of implementation in Skyrim for this. The click to sprint in FO4VR sustains for longer. Seems that there needs to be a timeout for the click to sprint so that it is held. 
Thats something I was thinking of anyways and can be done, just didnt need it before.

**Comment by BrentW89** [comment]
To clarify, the problem I’m having is not that it sprints when I don’t want it to. It is that it constantly stops when I keep moving (with or without the trackers). I’ve messed with tons of thresholds but this seems to happen randomly no matter how I set things. Read my latest post online with more details and let me know. Thanks!

**Comment by Blindmikey** [comment]
One issue I'm aware of in SkyrimVR is that items used to block, if equipped, can slow you down dramatically if they are moved in front of you, say if you're holding your shield in the left hand and while running, glance left. 

Not sure if that's related to your experience, but thought I'd throw that out there just in case.

**Comment by BrentW89** [comment]
No, it happens with nothing equipped. Let me know if it works correctly for
you.

**Comment by pottedmeat7** [comment]
@BrentW89 can you make a new issue with a exact description of what is happening, thank-you
